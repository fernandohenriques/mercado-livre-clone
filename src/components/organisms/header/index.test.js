import React from 'react';
import { shallow } from 'enzyme';
import { render } from '@testing-library/react';

import Header from './index';

describe('<Header />', () => {
  it('Snapshot testing', () => {
    const { asFragment } = render(<Header />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('Should find semantic tags Header and Figure', () => {
    const wrapper = shallow(<Header />);

    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.find('header')).toHaveLength(1);
    expect(wrapper.find('figure')).toHaveLength(1);
  });

  it('Should render SearchInput component with empty value', () => {
    const wrapper = shallow(<Header />);

    expect(wrapper.find('SearchInput')).toHaveLength(1);
    expect(wrapper.find('SearchInput').props().value).toBe('');
  });
});
