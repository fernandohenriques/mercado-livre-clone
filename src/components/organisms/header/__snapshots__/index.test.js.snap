// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Snapshot testing 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="container"
    >
      <div
        class="logo"
        role="button"
        tabindex="0"
      >
        <figure>
          <img
            alt="Mercado Livre - Onde comprar e vender de Tudo"
            src="logo.png"
            title="Mercado Livre - Onde comprar e vender de Tudo"
          />
        </figure>
      </div>
      <div
        class="container"
      >
        <input
          class="input input"
          placeholder="Buscar produtos..."
          type="text"
          value=""
        />
        <button
          type="button"
        >
          <img
            alt="Buscar"
            src="ic-search.png"
            title="Buscar"
          />
        </button>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
